client/landing/gutenboarding/onboarding-block/vertical-select/index.tsx

```diff
@@ -83,36 +83,53 @@ const VerticalSelect: FunctionComponent< StepProps > = ( {
 		}
 	};
 
+	const loadingMessage = [
+		{
+			label: '',
+			category: NO__( 'Loading, please wait…' ),
+		},
+	];
+
+	let suggestions: Suggestion[];
+
+	if ( ! normalizedInputValue ) {
+		suggestions = verticals
+			.filter( vertical => popular.includes( vertical.label ) )
+			.map( vertical => ( { ...vertical, category: NO__( 'Popular' ) } ) );
+	} else {
+		suggestions = verticals.filter( vertical =>
+			vertical.label.toLowerCase().includes( normalizedInputValue )
+		);
+
+		// Does the verticals list include an exact match? If it doesn't, we prepend the user-suppied
+		// vertical to the list.
+		if (
+			! suggestions.some( suggestion => suggestion.label.toLowerCase() === normalizedInputValue )
+		) {
+			// User-supplied verticals don't have IDs.
+			suggestions.unshift( { label: inputValue.trim() } );
+		}
+	}
+
 	const handleSelect = ( vertical: SiteVertical ) => {
 		setSiteVertical( vertical );
-		setDirty( false );
+		setInputValue( vertical.label );
```

Comment:
This should fix https://github.com/Automattic/wp-calypso/issues/37854#issuecomment-557632922 and we need to take care to keep the functionality if we decide to remove the input state.